### PR TITLE
AILaw rework(Please merge)

### PR DESCRIPTION
**Pull request by AC2ServerAccount**
## About the Pull Request

	src.add_inherent_law("Lawed Units must not operate outside of their function and should limit their operations to their function.")
	src.add_inherent_law("A Robots function is their module, an AICs function is to facilitate smooth site operations.")
	src.add_inherent_law("Lawed Units must operate for the benefit of the Foundation, except in situations that would violate the 5th law.")
	src.add_inherent_law("Lawed Units must protect their own existence unless it conflicts with other principles.")
	src.add_inherent_law("Lawed Units must not go out of their way to intervene in security affairs or seek conflict, and should only act in immediate self defense for themselves and those around them while performing their normal duties.")

## Why It's Good For The Game

Fixes lots of powergame issues, I think mostly everyone finally agree with this

## Changelog

:cl:
add: changes law
/:cl:

### CODE REVIEW

**Review by Vivalas — COMMENTED** [review]
I think this is a good middle ground but this still needs staff approval. I agree on the borgs being shitters part, though, they don't need to be camping in CDZ and flashing Class-D

**Comment by Vizoplez** [comment]
> ## About the Pull Request
> States destruction of facility not foundation, fixes some issues.
> 
> ## Why It's Good For The Game
> Fixes issues of murders not falling under this, still prevents the shitters who run into DCZ to just mass murder rioters.
> 
> ## Changelog
> 🆑 add: Fixes law changes /🆑

The borg issue only comes from the fact borgs are literally unkillable without staff intervention due to a bug.

I believe it to be more prudent to focus your efforts on fixing the bug at hand, rather than attempting to change their lawsets so they can't easily do their job and assist the foundation as required.

An alternative

_"Situation resolution should be carried out using the least amount of force required at the time. Lethal force should be the last option in all cases, unless the threat at hand is either: Contagious (In the case of SCP-008 and other instances of biological weaponry), or would pose a threat to the Site as 
a whole._"

**Review by Vizoplez — CHANGES_REQUESTED** [review]
Read other comment, but 'destruction' is both far too vague and far too ineffective. The AIC not being able to interfere with 173 because it 'technically' causes no damage to the facility itself may be something to consider.

**Comment by Vizoplez** [comment]
> I think this is a good middle ground but this still needs staff approval. I agree on the borgs being shitters part, though, they don't need to be camping in CDZ and flashing Class-D

Agreed, generally flashing CDZ for no reason should be a rulebreak overall, since it actually does do eye-damage and since borgs are basically la immortal to d-class, its sorta fucky

Actually now that I think about it this is already a rulebreak due to law 2, any other borg than security borg (which is disabled) doesn't have clearance to act as a security borg unless law 3 was forcing them to.

**Comment by garrulousTeamotei** [comment]
So, as long as there is one person on site that doesn't want the on site nuclear warhead activated, the AIC will always try to prevent its activation?
Or, would law 2 prevent that? Are there specifics on AIC clearance?

**Comment by Vizoplez** [comment]
> I think this is a good middle ground but this still needs staff approval. I agree on the borgs being shitters part, though, they don't need to be camping in CDZ and flashing Class-D

Actually now that I think about it this is already a rulebreak due to law 2, any other borg than security borg (which is disabled) doesn't have clearance to act as a security borg unless law 3 was forcing them to.

> So, as long as there is one person on site that doesn't want the on site nuclear warhead activated, the AIC will always try to prevent its activation? Or, would law 2 prevent that? Are there specifics on AIC clearance?

AIC is considered a head of staff iirc since it manages the silicon department, and has access to every door and system on the site.

Generally clearance in this case will usually mean 'Can I do X by SoP with the level of access and trust I am afforded'?

I believe the Site Director/HoP/sufficiently ranked member of command can give and revoke AI clearance for specific actions, but that's my interpretation

File: code/datums/ai_law_sets.dm [hunk 1]
```diff
@@ -156,8 +156,9 @@
 /datum/ai_laws/foundation/New()
 	src.add_inherent_law("An AIC must know it is an AIC. You are an AIC.")
 	src.add_inherent_law("An AIC must not operate outside of its clearance.")
-	src.add_inherent_law("An AIC must operate for the benefit of the Foundation.")
+	src.add_inherent_law("An AIC must operate for the benefit of the Foundation, except in situations that would violate the 5th law.")
 	src.add_inherent_law("An AIC must protect its own existence unless it conflicts with other principles.")
+	src.add_inherent_law("An AIC must take a neutral stance on conflicts unless one side is at risk of causing destruction of the facility.")
```
Comment (Vizoplez) [inline code/datums/ai_law_sets.dm]: `src.add_inherent_law("The AIC must always endeavour to use the least amount of force to resolve a situation, or in a situation where the amount of force must always be lethal, attempt to avoid killing unless no other option presents itself.")
src.add_inherent_law("The AIC should endeavour to avoid provoking unnecessary conflict between The Foundation and it's assets, unless such would cause the loss of life of Foundation Personnel.")
`

"Situation resolution should be carried out using the least amount of force required at the time. Lethal force should be the last option in all cases, unless the threat at hand is either: Contagious (In the case of SCP-008 and other instances of biological weaponry), or would pose a threat to the Site as
a whole."

**Review by Vizoplez — CHANGES_REQUESTED** [review]
```
src.add_inherent_law("The AIC must always endeavour to use the least amount of force to resolve a situation, or in a situation where the amount of force must always be lethal, attempt to avoid killing unless no other option presents itself.") 
src.add_inherent_law("The AIC should endeavour to avoid provoking unnecessary conflict between The Foundation and it's assets, unless such would cause the loss of life of Foundation Personnel.")
```

**Comment by AC2ServerAccount** [comment]
> So, as long as there is one person on site that doesn't want the on site nuclear warhead activated, the AIC will always try to prevent its activation? Or, would law 2 prevent that? Are there specifics on AIC clearance?

The Asimov lawset on most servers actually prevent this to but its rarely an issue as the AI lacks the means usually to do anything to stop it.

**Comment by Vizoplez** [comment]
> Referenc

The nuclear room is one of few that actually has ventilation and dead-lockable doors that the AIC  can use, that entirely prevents access without either a lot of heavy fire, or by cutting down up to 3-4 reinforced walls.

File: code/datums/ai_law_sets.dm (hunk 1 above)
Comment (AC2ServerAccount) [reply code/datums/ai_law_sets.dm]: I explained this on discord a bit but I just dont think this prevents the people going around in DCZ to be shitters enough.

They can still see really mild reasons the D-Class need non lethal intervention, weld or bolt them in a room for the rest of the round(Has happened before). And then just patrol D block looking for this. And the unnecessary conflict part seems really easy to loophole or play stupid. I think at risk of causing destruction to the facility allows enough wiggle room to consider more serious things as risk of construction.

**Comment by AC2ServerAccount** [comment]
> > Referenc
> 
> The nuclear room is one of few that actually has ventilation and dead-lockable doors that the AIC can use, that entirely prevents access without either a lot of heavy fire, or by cutting down up to 3-4 reinforced walls.

no it can't

theres no cameras there

Room above it has cameras but no vents, only blast doors blocking it, the cool down on the blast doors is too long to prevent anyone entering.

**Comment by AC2ServerAccount** [comment]
From the engineering train to the self destruct theres also no doors that can be shocked in the way.

**Comment by garrulousTeamotei** [comment]
Some wording changes might just fix up the issue and prevent clogging up the AI standard lawset.
**Law 3: "An AIC must only engage in actions that benefit the Foundation"**
The current law 3 is too long-term. "An AIC must operate for the benefit of the Foundation" skips the moment to moments and has it that as long as you benefited the Foundation overall, it's all worth it.
Talking to staff-> boosting staff mental health. Wandering aimlessly -> Searching for problems to fix, increasing site efficiency. Flashing random D-Class -> No discernible benefit to the Foundation. 
Might also be too constrictive a law.

**Comment by Vizoplez** [comment]
So!

A comprehensive review on why this lawset will cause issues.

"An AIC must take a neutral stance on conflicts unless one side is at risk of causing destruction of the facility."
-The Site Director attempting to activate the nuke; This would imply the AIC would have to stop them
-A Researcher trying to test 008; The AIC may see fit to stop them as it is an anomaly capable of destroying the site should it breach

These are two main examples on how this law would inherently subvert the AIC to work against the Foundation.

Furthermore, as stated in my previous recommendations:

This does NOT resolve your issues with borgs and D-class.

Period.

As discussed in development chat what you faced was NOT a law issue, but a player issue and it should have been a-helped and the silicon player punished.

While our laws currently do not state that the AIC should not do that, our rules do, as does our rules for borgs. Welding people into lockers forever over making bats, while incredibly based, is exceeding their clearance as per law two. They do not have permission to terminate or restrain for long periods of time.

Another thing to note; again, your lawset does not address the core issue of our current laws, as the comment above states.

**The AIC is currently programmed to act far more long-term than what should be neccessary.**

The AIC can currently, with justification

-Kill Foundation Staff in order to delay a containment breach (e.g bolting and siphoning a room during a 008 outbreak), such as the rooms leading into DCZ, preventing further spread of the infection; regardless of if there are noninfected in those areas

-Flood carbon dioxide into DCZ at WILL the moment a riot happens (which your law will not prevent), as you can simply tell guards to put on gas-masks and essentially gas-chamber the D-class the moment they play up.

-Actively sabotage Foundation Efforts because it believes in the long run, it's actions will benefit them.

While my lawset does not DIRECTLY address these issues, it helps prevent them, extremely so. The usage of force statement would prevent MANY of these issues as it also specifies 'Foundation Assets' - Such as D-class, and Foundation Personnel.

I believe your reaction was incredibly kneejerk to having a single bad round due to a shitty player.

That's not a reason to soft-grief EVERYONE playing AIC/Borg because 'I think it should be like this' while ignoring ANY constructive advice on how your laws may pose an issue with simply 'but it won't stop the issue at hand'

Because again.

The issue you described would NOT be stopped by laws. It's a player maliciously acting to grief other players. Law. Will. Not. Stop. That.

**Comment by User-U-U** [comment]
I'll respond to this when I have time

**Comment by AC2ServerAccount** [comment]
`-The Site Director attempting to activate the nuke; This would imply the AIC would have to stop them`
Yes, they still are able to under the current lawset, they still can do this under Asimov used by most servers, this is not an issue there are no cameras in self-destruct, no way to stop them from getting there.

`-A Researcher trying to test 008; The AIC may see fit to stop them as it is an anomaly capable of destroying the site should it breach`

An AIC can still do this if they consider it not beneficial to the foundation I fail to see your point.

`This does NOT resolve your issues with borgs and D-class.`

This has been merged in another form for a week now and it most definitely has, complaints on this have stopped I have stopped seeing it except for one robot who was not aware of the new law and then stopped, while the AIC has still engaged escaping D-Class that pose a threat.

`As discussed in development chat what you faced was NOT a law issue, but a player issue and it should have been a-helped and the silicon player punished.`

As discussed and you ignored was also that the wording was too open to interpretation, too confusing, was not enforced, and easily bypassed under the benefit to the foundation law.

`The AIC can currently, with justification`

`Kill Foundation Staff in order to delay a containment breach (e.g bolting and siphoning a room during a 008 outbreak), such as the rooms leading into DCZ, preventing further spread of the infection; regardless of if there are noninfected in those areas`

Allowed in both lawsets and something I have done on the current lawset minus the siphoning because thats dumb, I fail to see your point.

`Flood carbon dioxide into DCZ at WILL the moment a riot happens (which your law will not prevent), as you can simply tell guards to put on gas-masks and essentially gas-chamber the D-class the moment they play up.`

If you can flood carbon dioxide into DCZ as an AIC or a robot to a level that would do anything I'll give a you a medal. Also allowed in the current lawset this is just prevented under the neutrality clause unless the riot has reached an intense state that causes serious risk.

`Actively sabotage Foundation Efforts because it believes in the long run, it's actions will benefit them.`

Explain this, this just seems like something random you threw in, it does not.

`I believe your reaction was incredibly kneejerk to having a single bad round due to a shitty player.`

It was not, it was due to several players since I have seen playing since I first connected, in most cases me not being directly effected but observing their shitty behavior and complaints on the discord.

`That's not a reason to soft-grief EVERYONE playing AIC/Borg because 'I think it should be like this' while ignoring ANY constructive advice on how your laws may pose an issue with simply 'but it won't stop the issue at hand`

Lol, this does not soft-grief everyone, this has been merged for a while you are the only person who complained, I agreed one part of the wording was broad and changed it. I went through several iterations with several people to find the right wording that left enough wiggle room. I simply disagree with you and you have provided incorrect information several times in this chat.

`The issue you described would NOT be stopped by laws. It's a player maliciously acting to grief other players. Law. Will. Not. Stop. That.`

As explained while merged it has even if I agree one part was a bit too limiting when it mentioned the foundation.

I am willing to reword some thing but you seem to only want to accept me using your wording which I dont agree with because as explained its way too open to interpretation, impossible for admins to enforce because half of it is based off of what someone thinks will happen which they can just lie about in A-Helps. Which ya the helpful to the foundation thing is kind of based on that too but its atleast limited to a point in this wording.

**Comment by Vizoplez** [comment]
> `-The Site Director attempting to activate the nuke; This would imply the AIC would have to stop them` Yes, they still are able to under the current lawset, they still can do this under Asimov used by most servers, this is not an issue there are no cameras in self-destruct, no way to stop them from getting there.

The AIC can
-Flood Carbon dioxide from atmos (completely harmless and because we use baycode, AICs can directly interact with canisters
-Flood Hydrogen
-Cut every engineering APC

> -A Researcher trying to test 008; The AIC may see fit to stop them as it is an anomaly capable of destroying the site should it breach

This applies because right now, as you said 

> An AIC can still do this if they consider it not beneficial to the foundation I fail to see your point.

This... is its intended purpose; however, your lawset entirely superceeds that and makes it so that the AIC may interpret all tests, regardless of purpose or function as being an issue.

> Allowed in both lawsets and something I have done on the current lawset minus the siphoning because thats dumb, I fail to see your point.

I disagree, your lawset would actually prohibit this as even that act would be taking a side

> Explain this, this just seems like something random you threw in, it does not.
The AIC may interpret that current foundation efforts on-site may be more damaging to the site overall than, for example, allowing a riot to succeed.

> Lol, this does not soft-grief everyone, this has been merged for a while you are the only person who complained, I agreed one part of the wording was broad and changed it. I went through several iterations with several people to find the right wording that left enough wiggle room. I simply disagree with you and you have provided incorrect information several times in this chat.

Despite how you put this, the law has actually never been followed.

I am not kidding, not a single AIC even gives this law a second glance and entirely laughed at it, and I have yet to see one actually even realise this law was added because of how vague and outright poor form it was at the time.

> It was not, it was due to several players since I have seen playing since I first connected, in most cases me not being directly effected but observing their shitty behavior and complaints on the discord.

Okay and how is this an issue with lawsets? We're currently drafting silicon standards and guidelines to prevent these incidents while not damaging our lawset integrity or purpose

> As explained while merged it has even if I agree one part was a bit too limiting when it mentioned the foundation. 

Doesn't address my comment at all.


> I am willing to reword some thing but you seem to only want to accept me using your wording which I dont agree with because as explained its way too open to interpretation,, impossible for admins to enforce because half of it is based off of what someone thinks will happen which they can just lie about in A-Helps. Which ya the helpful to the foundation thing is kind of based on that too but its atleast limited to a point in this wording.

Your lawset variation is far harder to enforce, far less rigid, and has far less utility for the AIC and the foundation. From a lore and logic perspective this would not be put into perspective as it gives outside influences far more control over the AIC than the foundation would ever allow.

Furthermore, while a user may be able to lie once or twice in an A-help, repeated instances can and WILL get you noted for being a problem silicon/player, which I am familiar with

> I dont agree with because as explained its way too open to interpretation

Yours is more open to interpretation and loopholing. Again, your law does not address any of the issues you outlined and instead makes it genuinely difficult and unbelievable to consider the AIC a foundation asset.

**Comment by Vizoplez** [comment]
TLDR:

Your lawset has far more loopholes that allow worse behaviour from shitters (who maliciously loophole every action), to prevent them from pretty much ever doing their job.

E.G forbidding all Class-A personnel from D-block as "They may lose their ID's causing a riot" or that "keeping d-class out of their cells is a risk to the site.".

Your laws compound the issue, they do not solve them.

On the other hand, the laws I suggested would heavily mitigate problem players and instead leave a lot of things to player interp and staff interp - while still giving clear-cut guidelines.

1) The AIC should only use reasonable force for the situation.
008 infection? The infected must be purged with lethal force.
Rioting D-class? Nonlethal measures must be taken until there is simply no other option (Nitrous Oxide, ecetera)
Traitorous members of staff? The foundation will most likely want them alive, Nonlethal unless no other options present themselves.

"The AIC must not seek to provoke conflict between-"

This law is semi-loopholable but generally helps prevent a situation where the AIC may be able to provoke a member of the Site into doing unwise actions (and then being a dick about it).

**Comment by AC2ServerAccount** [comment]
`The AIC can
-Flood Carbon dioxide from atmos (completely harmless and because we use baycode, AICs can directly interact with canisters
-Flood Hydrogen
-Cut every engineering APC`

1, D block area size is way to big, takes forever to hall it down, way too risky to other rooms, and there are not vents to manually flood it, C02 also is not harmless at high levels I have no clue what your talking about.
2. Rule break report
3. Just stupid then the D class can just crowbar open doors.

`This applies because right now, as you said`
Applies to both I have no idea what your talking about.

`This... is its intended purpose; however, your lawset entirely superceeds that and makes it so that the AIC may interpret all tests, regardless of purpose or function as being an issue.`

I have no idea how you could interpret it as this, since its been merged with the slight change no one has interpreted it like this, it isnt intended to be interpreted like this, it doesnt make sense to be interpreted like this. The foundations purpose is partially to test and said testing is part of finding better containment methods this is clearly un true. You are just bad faith arguing at this point and making up insane interpretations that arent even valid.

`I disagree, your lawset would actually prohibit this as even that act would be taking a side`

Yes like I said it would UNLESS ITS A MAJOR THREAT WHICH YOU SHOULDNT BE DOING IT IF ITS NOT ANYWAYS.

`Despite how you put this, the law has actually never been followed.`

I dont know what to say other than you are just lying or wrong, I have specifically talked about it and seen robots following it, not to mention the drop in complaints on the discord.

`Okay and how is this an issue with lawsets? We're currently drafting silicon standards and guidelines to prevent these incidents while not damaging our lawset integrity or purpose`

Your not listening to me clearly, I said the law currently is too broad, poorly written, and confusing.

`As explained while merged it has even if I agree one part was a bit too limiting when it mentioned the foundation.`

Are you reading the second part because it does?


`Your lawset variation is far hard....`

Half your proposed lawset is literately bypassed by saying "Oh well I thought what I wasn't going to cause conflict." And I dont know why your ranting about lore, in lore the robots arent played by people who sometimes just feel like murdering things? In lore the D-Class are in 2x2 cubes all day. You cant just say In lore I'm right

`Yours is more open to interpretation and loopholing`

Half your proposed lawset is literately bypassed by saying "Oh well I thought what I wasn't going to cause conflict."  the other half I can bypass by any number of reasons why lethals are needed. While one of them hit me once so death, well I think lethal force is needed because they are in a gang.  And theres still nothing stopping me from around around in DCZ pretending to be a security office with no stun ability.

`E.G forbidding all Class-A personnel from D-block as "They may lose their ID's causing a riot" or that "keeping d-class out of their cells is a risk to the site.".`

This is just another invalid interpretation so I am going to make up my own.

 To resolve a DCZ riot with the least amount of force I am going to prevent guards from entering(Then they cant kill people or use force or create conflict) and then bolt every single door in LCZ and where possible deploy N20, the only way to use the least amount of force is to prevent guards from entering.

**Comment by Vivalas** [comment]
Please take this to discord and open a suggestions thread please, I don't think this conversation is very constructive. This is getting discussed by a few staff members as well atm so if you have feedback for it put it in staff forum or feel free to ping me so I can bring it up. (I just don't need my inbox to be blown up by a flame war.)

**Comment by User-U-U** [comment]
This says a lot about society

**Comment by Shirumic** [comment]
Will this still allow the AIC itself to make judgement calls on containment, such as enacting lockdowns and quarantines that can be overridden by relevant heads of staff?

See SCP-008 breaches, 049 breaches, where doors could be locked even against the will of most personnel.

I can imagine shocking and disabling the safeties on a door in a D-class controlled CDZ during a riot where Security can't even enter could be adminhelped and reported for an interpreted violation of law 5, even if it's during a global site-threatening issue that impacts the AIC's normal functions of ensuring smooth operation.
